File: ChatAppServer/Schema/Queries/GroupQuery.cs
using ChatAppServer.Repositories.IRepositories;
using ChatAppServer.Schema.Types;

namespace ChatAppServer.Schema.Queries
{
    [ExtendObjectType(typeof(Query))]
    public class GroupQuery
    {
        public async Task<List<GroupType>?> GetGroupsOfUser(string ownerId, [Service] IGroupRepository groupRepository)
        {
           var groups = await groupRepository.GetGroupsOfUser(ownerId);

            return groups.Select(g => new GroupType
            {
                GroupName = g.GroupName,
                Id = g.Id,
                OwnerId = ownerId,
                Password = g.Password,
                UniqueCodeGroup = g.UniqueCodeGroup 
            }).ToList();
        }
    
    
        public async Task<List<string>> GetGroup(string uniqueCodeGroup, [Service] IGroupRepository groupRepository, [Service]IAuthRepository authRepository)
        {
            var group = await groupRepository.GetGroup(uniqueCodeGroup);

            List<string> memberNames = [];
            foreach (var member in group.Members)
            {
                var user = authRepository.GetUserByUniqueCode(member);
                memberNames.Add(user.FullName);
            }

            return memberNames;
        }
    }
}

File: ChatAppClient/src/app/graphql/queries/group.query.ts
import { gql } from 'apollo-angular';

export const GET_GROUPS_OF_USER = gql`
	query ($ownerId: String!) {
		groupsOfUser(ownerId: $ownerId) {
			id
			ownerId
			groupName
			uniqueCodeGroup
		}
	}
`;

export const GET_MEMBERS_OF_GROUP = gql`
	query ($uniqueCodeGroup: String!) {
		group(uniqueCodeGroup: $uniqueCodeGroup)
	}
`;
